WifiManager/impl/WifiManagerSignalThreshold.cpp

```diff
@@ -30,14 +30,41 @@ namespace {
     const float signalStrengthThresholdGood = -60.0f;
     const float signalStrengthThresholdFair = -67.0f;
 
+    #define BUFFER_SIZE 512
+    #define Command "wpa_cli signal_poll"
+
+    std::string retrieveValues(const char *command, char *output_buffer, size_t output_buffer_size)
+    {
+        std::string key, value;
+        std::string rssi = "";
+
+        FILE *fp = popen(command, "r");
+        if (!fp)
+        {
+            LOGERR("Failed in getting output from command %s \n",command);
```

Comment:
If popen fails, it must return from here..  
Otherwise, The below while loop end up in accessing the null pointer and the pclose(NULL) etc..